Backend/Backend.Tests/SubscriptionTest.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Backend.Controllers;
+using Backend.Controllers.Requests;
+using Backend.Controllers.Responses;
+using Backend.Models;
+using Backend.Services;
+using Backend.Services.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.VisualBasic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MockQueryable.Moq;
+using Moq;
+using Assert = Xunit.Assert;
+using Xunit;
+
+
+namespace Backend.Tests
+{
+    [Collection("Sequential")]
+    public class SubscriptionTest : IAsyncLifetime
+    {
+        private Mock<AppDbContext> _dbContextMock;
+        private Mock<IAuthService> _authServiceMock;
+        private Mock<IProfileService> _profileServiceMock;
+        private Mock<HttpContext> _httpContextMock;
+        
+        private SubscriptionService _subscriptionService;
+        private SubscriptionController _subscriptionController;
+
+        private const string DOMAIN = "TestDomain";
+
+        public SubscriptionTest()
+        {
+            // Re-initilize every test
+            _dbContextMock = new(new DbContextOptions<AppDbContext>());
+            _authServiceMock = new();
+            _profileServiceMock = new();
+            _subscriptionService = new(_dbContextMock.Object);
+            _httpContextMock = new();
+            _subscriptionController = new(_authServiceMock.Object, _subscriptionService)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = _httpContextMock.Object
+                }
+            };
+
+            MockBasicUtilities(out var podcast, out var user, out var episode);
+        }
+      
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        public Task InitializeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            // Re-initilize every test
+            _dbContextMock = new(new DbContextOptions<AppDbContext>());
+            _authServiceMock = new();
+            _profileServiceMock = new();
+            _subscriptionService = new(_dbContextMock.Object);
+            _httpContextMock = new();
+            _subscriptionController = new( _authServiceMock.Object,_subscriptionService)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = _httpContextMock.Object
+                }
+            };
+
+            MockBasicUtilities(out var podcast, out var user, out var episode);
+        }
+
+        [Fact]
+        public void Subscribe_Podcast_Success()
```

Comment:
Make your test titles a little more specific. Should be something like this
Subscription_.SubscribeAsync_ValidRequest_Success 
in other words:
What feature you are testing, what function you are testing, what you are testing about it, and what the expected result is

Reply:
makes it easier to search for this, and when it fails its quicker to notice what is wrong

Reply:
redid all tests. Should be fine